doc/10-Channels.md

```diff
@@ -1,88 +1,298 @@
 
-To make those plugins available to Icinga Notifications, they must be placed in the
+To make those channels available to Icinga Notifications, they must be placed in the
 [channels directory](03-Configuration.md#channels-directory),
 which is being done automatically for package installations.
-Afterwards they can be configured through Icinga Notifications Web.
+At startup, Icinga Notifications scans this directory, starts each channel once to query its configuration options
+and stores these options in the database.
+Using this information, Icinga Notifications Web allows channels to be configured,
+which are then started, configured, and lastly used to send notification events from Icinga Notifications.
 
 ## Technical Channel Description
 
-Channel plugins are processes that run continuously and independently of each other. They receive many requests over
-their lifetime. They receive JSON-formatted requests on stdin and reply with JSON-formatted responses on stdout. The
-request and response structure is inspired by JSON-RPC.
+!!! warning
+
+    As this is still an early preview version, things might change.
+    There may still be severe bugs and incompatible changes may happen without any notice.
+
+Channel plugins are independent processes that run continuously, supervised by Icinga Notifications.
+They receive JSON-formatted requests on `stdin` and reply with JSON-formatted responses on `stdout`.
+The request and response structure is inspired by JSON-RPC.
+
+Note that this documentation uses beautified JSON for ease of reading.
+
+For logging or debugging, channels can write to `stderr`, which will be logged by Icinga Notifications.
 
 ### Request
 
-The request must be in JSON format and should contain following keys:
+A channel receives a request as a JSON object with the following fields:
 
 - `method`: The request method to call.
-- `params`: The params for request method.
+- `params`: Optional params for the request method.
 - `id`: Unsigned int value. Required to assign the response to its request as responses can be sent out of order.
 
 Examples:
 
+- Simple request without any `params`:
+  ```json
+  {
+    "method": "Simple",
+    "id" : 1000
+  }
+  ```
+- Request with `params` of different types:
+  ```json
+  {
+    "method": "WithParams",
+    "params": {
+      "foo": 23,
+      "bar": "hello"
+    },
+    "id": 1000
+  }
+  ```
+
+### Response
+
+Each request must be answered by the channel with a response JSON object of the following fields:
+
+- `result`: The result as JSON format. Omitted when the method does not return a value, i.e., for setter calls, or an
+  error has occurred.
+- `error`: The error message. Omitted when no error has occurred.
+- `id`: The request id.
+
+In case of a present `error` value, the `result` field should be empty.
+Successfully responses without a `result` do only contain the `id` field.
```

Comment:
This (few other places as well) should be rephrased as [`Successful`](https://dictionary.cambridge.org/dictionary/english/successful) instead.